### PR TITLE
fix: stored procedures are not updatable due to a missing boolean = true

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

**Comment by lmsurpre** [comment]
Seems fine to me.  A related question I have is whether it updates all FHIR_ADMIN schema stored procs (e.g. setTenant) or just the schema that is passed in.
The reason I ask is that our docs currently say "Update the stored procedures and functions for FHIRDATA (and not FHIR_ADMIN)" but I'm not sure that is right.
If it doesn't update FHIR_ADMIN stored procs, do we have a different means for updating that one?

**Comment by prb112** [comment]
> Seems fine to me. A related question I have is whether it updates all FHIR_ADMIN schema stored procs (e.g. setTenant) or just the schema that is passed in.
> The reason I ask is that our docs currently say "Update the stored procedures and functions for FHIRDATA (and not FHIR_ADMIN)" but I'm not sure that is right.
> If it doesn't update FHIR_ADMIN stored procs, do we have a different means for updating that one?

so ... it turns out we never supported that
we just created the set_tenant and only updated with the --update-schema

**Comment by punktilious** [comment]
This must be a regression because I used to update the stored procedures all the time when they were first being developed. In fact, the update option was added so that I could quickly update just the stored procs without having to process the rest of the model.

**Comment by punktilious** [comment]
Also, I would've preferred to keep the various schemas in different PhysicalDataModel instances. I've considered a refactor of that stuff, but too much other stuff going on at the moment.

**Comment by prb112** [comment]
> This must be a regression because I used to update the stored procedures all the time when they were first being developed. In fact, the update option was added so that I could quickly update just the stored procs without having to process the rest of the model.

Yes, it's a regression

**Review by lmsurpre — APPROVED** [review]
LGTM

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/BaseObject.java [hunk 1]
```diff
@@ -255,7 +255,7 @@ public void applyTx(IDatabaseAdapter target, ITransactionProvider tp, IVersionHi
      */
     @Override
     public void applyVersion(IDatabaseAdapter target, IVersionHistoryService vhs) {
-        if (vhs.applies(getSchemaName(), getObjectType().name(), getObjectName(), version)) {
+        if (vhs.applies(getSchemaName(), getObjectType().name(), getObjectName(), version) || getObjectType().equals(DatabaseObjectType.PROCEDURE)) {
```
Comment (lmsurpre) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/BaseObject.java:258]: wonder if we should add a comment to explaining the why.
basically we just want the procedures to get re-written every time (whether its a newer version or not)?

Comment (prb112) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/BaseObject.java:258]: Right.  I can add a comment

**Pull request merged**